core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -872,6 +917,387 @@ export class ExplorationEditor extends BaseUser {
+        throw new Error(
+          `Revision ${revisionNumber} is missing or has an empty ${property}`
+        );
+      }
+
+      showMessage(`Revision ${revisionNumber} has the property ${property}`);
+    }
+  }
+
+  /**
+   * Checks if the revisions are ordered by a specific property.
+   * @param {string} property - The property to check the order by.
+   */
+  async expectRevisionsToBeOrderedBy(property: string): Promise<void> {
+    if (!property) {
+      throw new Error('Invalid property');
+    }
+
+    let revisions = await this.page.$$(historyListItem);
+    for (let i = 0; i < revisions.length - 1; i++) {
+      let value1 = await revisions[i].$eval(property, el => el.textContent);
+      let value2 = await revisions[i + 1].$eval(property, el => el.textContent);
+
+      if (!value1 || !value2) {
+        throw new Error(`Failed to retrieve property ${property}`);
+      }
+
+      if (typeof value1 === 'string') {
+        value1 = value1.toString();
+        value2 = value2.toString();
+      }
+
+      if (value1 < value2) {
+        throw new Error(`Revisions are not sorted by ${property}`);
+      }
+    }
+
+    showMessage(`Revisions are sorted by ${property}`);
+  }
+
+  /**
+   * Filters the revisions by a specific user.
+   * @param {string} userName - The name of the user to filter the revisions by.
+
+   */
+  async filterRevisionsByUser(userName: string): Promise<void> {
+    if (!userName) {
+      throw new Error('Invalid user name');
+    }
+
+    await this.type(userNameEdit, userName);
+    // The unicode '\u000d' is used to simulate the Enter key press.
+    await this.type(userNameEdit, '\u000d');
+  }
+
+  /**
+   * Checks if the number of revisions matches the expected number.
+   * @param {number} expectedNumber - The expected number of revisions.
+   */
+  async expectNumberOfRevisions(expectedNumber: number): Promise<void> {
+    if (!Number.isInteger(expectedNumber) || expectedNumber < 0) {
+      throw new Error('Invalid expected number of revisions');
+    }
+
+    let revisions = await this.page.$$(historyListItem);
+    if (revisions.length !== expectedNumber) {
+      throw new Error(
+        `Expected ${expectedNumber} revisions, but found ${revisions.length}`
+      );
+    } else {
+      showMessage(`Found ${expectedNumber} revisions as expected`);
+    }
+  }
+
+  /**
+   * Adjusts the paginator to show a specific number of revisions per page.
+   * @param {number} revisionsPerPage - The number of revisions to show per page. Can be 10, 15, or 20.
+   */
+  async adjustPaginatorToShowRevisionsPerPage(
+    revisionsPerPage: number
+  ): Promise<void> {
+    const REVISIONS_PER_PAGE_OPTIONS: {[key: number]: number} = {
+      10: 1,
+      15: 2,
+      20: 3,
+    };
+    const index = REVISIONS_PER_PAGE_OPTIONS[revisionsPerPage];
+    if (index === undefined) {
+      throw new Error(
+        `Invalid number of revisions per page: ${revisionsPerPage}`
```

Comment:
Change error message to `${revisionsPerPage} is not one of the valid number of revisions per page: ${Object.keys(REVISIONS_PER_PAGE_OPTIONS)}`.